### ISSUE TITLE
Update controller-runtime and k8s to the latest version

### ISSUE THREAD

**Issue by marquiz** [kubernetes-sigs/node-feature-discovery-operator#208]
**What would you like to be added**:

Update `sigs.k8s.io/controller-runtime` and `k8s.io/*` dependencies to the latest versions (controller runtime v0.16 and k8s v1.28). We currently depend on k8s v1.26 which will be EOL'd around February 2024.

**Why is this needed**:

Keep up with the development. Avoid outdated deps.

**Comment by marquiz** [comment]
@yevgeny-shnaidman would you be interested in working on this?

**Comment by yevgeny-shnaidman** [comment]
yes, if it can be postponed till the last week of October (2 weeks from now)

**Comment by marquiz** [comment]
Cool 💪 And yeah, sure, we're not in a hurry with this one. Just would be good to finish before K8s v1.26 reaches EOL

**Comment by ArangoGutierrez** [comment]
Awesome, let us know how can we help! always good to have more contributors to the NFD project

**Comment by ArangoGutierrez** [comment]
This could be fun, controller-gen is a PITA
